server/src/main/java/org/elasticsearch/cluster/coordination/LeaderChecker.java

```diff
@@ -159,11 +155,13 @@ private void handleLeaderCheck(LeaderCheckRequest request) {
         assert discoveryNodes != null;
 
         if (discoveryNodes.isLocalNodeElectedMaster() == false) {
-            logger.debug("non-master handling {}", request);
-            throw new CoordinationStateRejectedException("non-leader rejecting leader check");
+            logger.debug("rejecting leader check on non-master {}", request);
+            throw new CoordinationStateRejectedException(
+                "rejecting leader check from [" + request.getSender() + "] sent to a node that is not the master");
```

Comment:
following the same logic you've used for the `nodeExists` check below, should this say "is no longer the master"?